compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2281,6 +2281,121 @@ impl<'tcx> Ty<'tcx> {
             _ => None,
         }
     }
+
+    pub fn is_c_void(self, tcx: TyCtxt<'_>) -> bool {
+        match self.kind() {
+            ty::Adt(adt_def, ..) => {
+                let def_id = adt_def.0.did;
+                let crate_name = tcx.crate_name(def_id.krate);
+                if adt_def.is_enum()
+                    && tcx.item_name(def_id) == sym::c_void
+                    && (crate_name == sym::core
```

Comment:
Could you please introduce a lang item instead. Those are strongly preferred over matching item paths as they keep working even when items get moved and prevent accidentally applying to definitions from stable code. (core and std are valid crate names)

Reply:
Do you have an example or could give me some directions on how to do that? I tried my best to do it for a type but couldn't figure it out.

Reply:
Done.